scenarios/validation_receipts/src/main.rs

```diff
@@ -61,82 +58,81 @@ fn agent_setup(
 fn agent_behaviour(
     ctx: &mut AgentContext<HolochainRunnerContext, HolochainAgentContext<ScenarioValues>>,
 ) -> HookResult {
-    let reporter = ctx.runner_context().reporter();
-    let agent = ctx.get().cell_id().agent_pubkey().clone().to_string();
-
-    let action_hash: ActionHash = call_zome(
-        ctx,
-        "crud",
-        "create_sample_entry",
-        "this is a test entry value",
-    )?;
+    // check if pending action is empty, if so create a new entry
+    if ctx.get().scenario_values.pending_actions.is_empty() {
+        let action_hash: ActionHash = call_zome(
+            ctx,
+            "crud",
+            "create_sample_entry",
+            "this is a test entry value",
+        )?;
+
+        ctx.get_mut()
+            .scenario_values
+            .pending_actions
+            .insert(action_hash, HashMap::new());
+    }
 
-    ctx.get_mut()
+    // get the first pending action
+    let action_hash = ctx
+        .get()
         .scenario_values
         .pending_actions
-        .insert(action_hash, HashMap::new());
+        .keys()
+        .next()
+        .cloned()
+        .expect("Cannot be empty here");
 
+    // collect validation receipts until complete
     let start = Instant::now();
+    let wait_for_all = std::env::var_os("NO_VALIDATION_COMPLETE").is_none();
```

Comment:
With the env var enabled, I'm not clear it is collecting the most useful data. On every run through the scenario we're checking if we have received validation receipts for the oldest pending action. But we may receive validation receipts out-of-order from when the actions were authored. Should we instead check *all* pending actions to see if we have received validation receipts for any of them?

That said, it looks like the only deployments of the scenario currently have this env var disabled. Should we just remove it? Or should we add another deployment for running with the var enabled? Could be a follow up issue and PR.